### PR TITLE
Create the dropdown menu

### LINKED ISSUE #61: Create the dropdown menu
ℹ️ This issue is part of Epic #59 

## Objective

Create a dropdown menu with the transmission types according to the design: [Figma](https://www.figma.com/file/2SHaHIlfaR031SLnJPXgIR/OKdose?node-id=71%3A2772&t=vQ9mmEW7CYmzH4Zr-0)

image

## Checklist

- [ ] There are defined 3 dropdown menus: Transmitted by Vectors (not ETV), Zoonoses, and Mycobacteria. 
- [ ] There are the icons used in the design for the names in the menu
- [ ] There are the translation keys for English and Spanish
- [ ] For the transmitted by vector dropdown is associated Leishmaniasis disease
- [ ] The others transmitted types menu should not have any information yet.
- [ ] The menu has a responsive behavior (to use in mobile and desktop)

### PR DESCRIPTION
**Pull request by keliumJU**
## Description

Add dropdown menu according to the design
image

Resolves #61

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by jcamilocp — CHANGES_REQUESTED** [review]
Hil, I added some comments. Also, remove the wrong reference to the private module, please.

Thanks!

File: okdose-web/tests/DropdownMenu.test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    ).toBeDefined();
+
+    expect(screen.getByText('Transmitidas por Vectores')).toBeInTheDocument();
+    expect(screen.getByText('Zoonosis')).toBeInTheDocument();
+    expect(screen.getByText('Micobacterias')).toBeInTheDocument();
+    expect(screen.getByText('Leishmaniasis')).toBeInTheDocument();
+    expect(screen.getByText('Malaria')).toBeInTheDocument();
+    expect(screen.getByText('Enfermedad de Chagas')).toBeInTheDocument();
+  });
+});
```
Comment (LauraVilladakommit) [inline okdose-web/tests/DropdownMenu.test.js:101]: We should text the translations, not only if it is defined.

**Review by LauraVilladakommit — CHANGES_REQUESTED** [review]

File: okdose-web/assets/styles/main.css [hunk 2]
```diff
@@ -7,3 +7,11 @@ body,
 html {
   font-family: 'Poppins', sans-serif;
 }
+
+.collapsible .content {
+  padding: 6px;
+}
+.collapsible .header {
+  padding: 6px;
```
[the remark this answers is not in the document]
Comment (keliumJU) [reply okdose-web/assets/styles/main.css:15]: I'm going to change it to tailwind utility classes, with dynamic measures

File: okdose-web/tests/DropdownMenu.test.js (hunk 1 above)
Comment (keliumJU) [reply okdose-web/tests/DropdownMenu.test.js:81]: first we define that the translation key exists, then we check the burned text in the language we are testing.

```
    expect(
      screen.getByText(
        i18n.getDataByLanguage('en').translation.leishmaniasis.name
      )
    ).toBeDefined();
```

File: okdose-web/components/common/Section.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}) {
+  const config = {
+    defaultExpanded: defaultExpanded || false,
+    collapsedHeight: collapsedHeight || 0
+  };
+  const {getCollapseProps, getToggleProps, isExpanded} = useCollapse(config);
+
+  return (
+    <div className='px-5 w-full collapsible'>
+      <div className='header bg-white text-neutral-700' {...getToggleProps()}>
```
[the remark this answers is not in the document]
Comment (keliumJU) [reply okdose-web/components/common/Section.js:21]: you are right, change made

File: okdose-web/assets/styles/main.css [hunk 4]
```diff
@@ -7,3 +7,11 @@ body,
...
+  cursor: pointer;
+}
```
[the remark this answers is not in the document]
Comment (keliumJU) [reply okdose-web/assets/styles/main.css]: I made the change to `p-2` with the tailwind utility classes and the `cursor-pointer` is default.

File: okdose-web/components/common/Section.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+function Section ({
+  defaultExpanded,
+  collapsedHeight,
+  title,
+  children,
+  iconName,
+  iconRight
+}) {
+  const config = {
```
[the remark this answers is not in the document]
Comment (keliumJU) [reply okdose-web/components/common/Section.js:13]: we only need it for the dropdown, I think it's fine just there

File: okdose-web/components/common/Section.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
[the remark this answers is not in the document]
Comment (keliumJU) [reply okdose-web/components/common/Section.js]: the dropdown needs this configuration, as it is part of the **react-collapse** library
` <div className='header bg-white text-neutral-700' {...getToggleProps()}>`
and the spred operator is part of modern javascript

File: okdose-web/components/DropdownMenu.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import mycobacteriaIcon from '../assets/images/icons/mycobacteriaIcon.svg';
+import zoonosesIcon from '../assets/images/icons/zoonosesIcon.svg';
+import leftIcon from '../assets/images/icons/leftIcon.svg';
+import {useTranslation} from 'react-i18next';
+
+function DropdownMenu () {
+  const {t} = useTranslation();
+  return (
+    <div className='m-auto w-full sm:m-auto sm:w-[420px] sm:h-[180px] sm:z-10'>
+      <Section
```
[the remark this answers is not in the document]
Comment (keliumJU) [reply okdose-web/components/DropdownMenu.js]: this was refactored with the object of okdose/index.js

File: okdose/app/index.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import {
+  transmittedByVectors,
+  zoonoses,
+  mycobacteria
+} from '../transmission-types/';
+import byVectorsIcon from '../../okdose-web/assets/images/icons/byVectorsIcon.svg';
```
Comment (jcamilocp) [inline okdose/app/index.js:6]: Is this import absolutely necessary? Can't we export only text referencing the images here?

File: okdose-web/.eslintrc.json [hunk 9]
```diff
@@ -26,7 +26,7 @@
     "quotes": ["error", "single"],
     "no-duplicate-imports": ["error"],
     "react/react-in-jsx-scope": ["off"],
-    "react/jsx-props-no-spreading": ["error"],
+    "react/jsx-props-no-spreading": ["off"],
```
Comment (jcamilocp) [inline okdose-web/.eslintrc.json:29]: Why turn off this rule?

File: okdose-web/containers/Home.js [hunk 10]
```diff
@@ -42,11 +43,13 @@ function Home () {
           <Welcome />
         </>
       ) : (
-        <div className='flex flex-col justify-center items-center pl-5 pr-5 pb-0 gap-16'>
+        <div className='flex flex-col justify-center items-center'>
+          <h1 className='uppercase overflow-hidden text-lg'>Mycobactéries</h1>
```
Comment (jcamilocp) [inline okdose-web/containers/Home.js:47]: Is this text correct?

**Review by jcamilocp — COMMENTED** [review]
Code looks a lot cleaner! Added some comments

File: okdose/app/index.js (hunk 8 above)
Comment (keliumJU) [reply okdose/app/index.js:6]: The closest thing to change the import according to the parcel documentation would be this: 

const transmissionTypes = {
  byVectors: {
    name: transmittedByVectors.name,
    icon: new URL('../../okdose-web/assets/images/icons/byVectorsIcon.svg', import.meta.url),

File: okdose-web/containers/Home.js (hunk 10 above)
Comment (keliumJU) [reply okdose-web/containers/Home.js:47]: my error, it stayed when with the tests of the capital letters, I already remove it.

File: okdose/app/index.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  transmittedByVectors,
+  zoonoses,
+  mycobacteria
+} from '../transmission-types/';
+
+const transmissionTypes = {
+  byVectors: {
+    name: transmittedByVectors.name,
+    icon: 'byVectorsIcon',
+    types: {
```
Comment (jcamilocp) [inline okdose/app/index.js:11]: What are these `types` for? I think we should not access that deep here

Comment (keliumJU) [reply okdose/app/index.js:11]: `types` refers to the types of diseases, and inside it refers to the parent objects of each disease: leishmaniasis, chagas and malaria.

**Review by jcamilocp — APPROVED** [review]
LGTM!

**Review by LauraVilladakommit — APPROVED** [review]

**Pull request merged**